clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4853,6 +4853,10 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
         // header file.
         CmdArgs.push_back("-dependency-filter");
         CmdArgs.push_back(Args.MakeArgString(Header));
+
+        // Since this is host compilation and integration header is included,
+        // enable integration header based diagnostics.
```

Comment:
```suggestion
        // Since this is a host compilation and the integration header is included,
        // enable the integration header based diagnostics.
```